casepro/utils/__init__.py

```diff
@@ -80,6 +81,42 @@ def normalize(text):
     return unicodedata.normalize('NFKD', re.sub(r'\s+', ' ', text.lower()))
 
 
+def normalize_urn(urn):
+    """
+
+    """
+    scheme, path = urn.split(':', 1)
+    if scheme == 'tel':
+        (path, _) = normalise_tel_number(path)
+
+    return '%s:%s' % (scheme, path)
+
+
+def normalise_tel_number(number):
+    # remove other characters
+    number = re.sub('[^0-9\+]', '', number.lower())
+
+    # add on a plus if it looks like it could be a fully qualified number
+    if len(number) >= 11 and number[0] not in ['+', '0']:
+        number = '+' + number
+
+    normalized = None
+    try:
+        normalized = phonenumbers.parse(number, None)
+    except Exception:
```

Comment:
rather catch `phonenumbers.phonenumberutil.NumberParseException` here instead, `Exception` is too liberal to be useful.